File: app/src/main/java/com/nutdiary/diary/presenter/LoginPresenter.java
package com.nutdiary.diary.presenter;

import android.support.annotation.NonNull;

import com.nutdiary.diary.base.BasePresenter;
import com.nutdiary.diary.base.ResultBean;
import com.nutdiary.diary.bean.MainListBean;
import com.nutdiary.diary.contract.LoginContract;
import com.nutdiary.diary.model.HomeModel;
import com.nutdiary.diary.model.LoginModel;
import com.trello.rxlifecycle2.LifecycleProvider;
import com.trello.rxlifecycle2.android.ActivityEvent;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.observers.DefaultObserver;
import io.reactivex.schedulers.Schedulers;

public class LoginPresenter extends BasePresenter{
    private LoginContract.LoginModel loginModel;
    private LoginContract.LoginView loginView;
    public LoginPresenter(LoginContract.LoginView loginView,LifecycleProvider<ActivityEvent> provider) {
        super(provider);
        this.loginView = loginView;
        loginModel = new LoginModel();
    }
    public void userLogin(String phone,String pass){
        loginModel.userLogin(phone,pass)
                .subscribeOn(Schedulers.io()) // 在子线程中进行Http访问
                .observeOn(AndroidSchedulers.mainThread()) // UI线程处理返回接口
                .compose(getProvider().bindUntilEvent(ActivityEvent.DESTROY))// onDestroy取消订阅
                .subscribe(new DefaultObserver<ResultBean>() {  // 订阅
                    @Override
                    public void onNext(@NonNull ResultBean resultBean) {
                        if (resultBean.getCode()==1){
                            //jump

                        }
                        loginView.showToast(resultBean.getMsg());
                    }

                    @Override
                    public void onError(@NonNull Throwable e) {
                        loginView.showToast(e.toString());
                    }

                    @Override
                    public void onComplete() {

                    }
                });
    }
}

File: app/src/main/java/com/nutdiary/diary/presenter/LoginPresenter.kt
package com.nutdiary.diary.presenter

import com.nutdiary.diary.base.BasePresenter
import com.nutdiary.diary.base.ResultBean
import com.nutdiary.diary.bean.LoginResultBean
import com.nutdiary.diary.contract.LoginContract
import com.nutdiary.diary.localData.UserData
import com.nutdiary.diary.model.LoginModel
import com.trello.rxlifecycle2.LifecycleProvider
import com.trello.rxlifecycle2.android.ActivityEvent

import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.observers.DefaultObserver
import io.reactivex.schedulers.Schedulers

class LoginPresenter(private val loginView: LoginContract.LoginView, provider: LifecycleProvider<ActivityEvent>) : BasePresenter(provider) {
    private val loginModel: LoginContract.LoginModel

    init {
        loginModel = LoginModel()
    }

    fun userLogin(phone: String, pass: String) {
        loginModel.userLogin(phone, pass)
                .subscribeOn(Schedulers.io()) // 在子线程中进行Http访问
                .observeOn(AndroidSchedulers.mainThread()) // UI线程处理返回接口
                .compose(provider.bindUntilEvent(ActivityEvent.DESTROY))// onDestroy取消订阅
                .subscribe(object : DefaultObserver<LoginResultBean>() {  // 订阅
                    override fun onNext(resultBean: LoginResultBean) {
                        if (resultBean.code == 0) {
                            UserData.saveUserUUID(resultBean.data)
                            loginView.jumpMain()
                        }
                        loginView.showToast(resultBean.msg)
                    }

                    override fun onError(e: Throwable) {
                        loginView.showToast(e.toString())
                    }

                    override fun onComplete() {

                    }
                })
    }
}
